tests/testing-resources/include/aws/testing/mocks/aws/client/MockAWSClient.h

```diff
@@ -41,8 +43,14 @@ class AmazonWebServiceRequestMock : public Aws::AmazonWebServiceRequest
     virtual const char* GetServiceRequestName() const override { return "AmazonWebServiceRequestMock"; }
     virtual bool HasEmbeddedError(Aws::IOStream& body, const Aws::Http::HeaderValueCollection& header) const override {
         (void) header;
+        
+        auto readPointer = body.tellg();
```

Comment:
This fixes the body of response parsing issue existing in test cases. The pointer in the stream was being moved to the end which caused subsequent parsing to fail.